### PR TITLE
docs(aio): add content migrated from angular.io

### PR DESCRIPTION
**Pull request by petebacondarwin**
This content was generated by a tool: https://github.com/petebacondarwin/aio-migrator

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: aio/transforms/examples-package/services/region-matchers/html.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // These kind of comments are used in HTML
 module.exports = {
-  regionStartMatcher: /^\s*<!--\s*#docregion\s*(.*)\s*-->\s*$/,
-  regionEndMatcher: /^\s*<!--\s*#enddocregion\s*(.*)\s*-->\s*$/,
-  plasterMatcher: /^\s*<!--\s*#docplaster\s*(.*)\s*-->\s*$/,
+  regionStartMatcher: /^\s*<!--\s*#docregion\s*(.*?)\s*(?:-->)?\s*$/,
```
Comment (gkalpak) [inline aio/transforms/examples-package/services/region-matchers/html.js]: Why is this needed?

Comment (petebacondarwin) [inline aio/transforms/examples-package/services/region-matchers/html.js]: Because it turns out that some authors want to mark up the start of a comment with a doc-region. See

https://github.com/angular/angular.io/blob/master/public/docs/_examples/upgrade-phonecat-2-hybrid/ts/index.html#L34

I had meant to put an explanation in the commit message but I seem to have missed it.

https://github.com/angular/angular/pull/14604/commits/774a7b5256ac49b162609b5e1c0015bc14e19776#diff-57d521021abd250f46526ffceae1d8bfR39

Comment (gkalpak) [inline aio/transforms/examples-package/services/region-matchers/html.js]: I see. So it's not like the `docregion` comment  will be closed on another line. It will be closed with the `enddocregion` comment.

Comment (petebacondarwin) [inline aio/transforms/examples-package/services/region-matchers/html.js]: Right, which is really weird but that is the way the example is written. The problem is that the author wants the example in the docs to look one way but the file to act (in real life) in another way.

File: aio/transforms/examples-package/services/region-parser.js [hunk 2]
```diff
@@ -106,7 +106,7 @@ function regionParserImpl(contents, fileType) {
 }
 
 function getRegionNames(input) {
-  return input.split(',').map(name => name.trim()).filter(name => name.length > 0);
+  return (input.trim() === '') ? [] : input.split(',').map(name => name.trim());
```
Comment (gkalpak) [inline aio/transforms/examples-package/services/region-parser.js]: Shouldn't this be  `(input.trim() === '') ? [''] : ...`.

**Comment by gkalpak** [comment]
I haven't looked at the two commits that bring content over from angular.io:

- e9c1852 - docs(aio): migrate content docs fromn angular.io   <-- _There's a typo there btw._
- b8bfca5 - feat(aio): copy example code from angular.io

The rest LGTM.

File: aio/transforms/examples-package/services/region-parser.js (hunk 2 above)
Comment (petebacondarwin) [inline aio/transforms/examples-package/services/region-parser.js]: This is covered in the regionParser, where there are some lines that say:

```js
        if (regionNames.length === 0) {
          regionNames.push('');
        }
```

I accept that it could probably be refactored at some point. But rest assured that it works.

Comment (petebacondarwin) [inline aio/transforms/examples-package/services/region-parser.js]: Ahah! Actually, you can see that we need to special case when there are no named regions. See https://github.com/petebacondarwin/angular/blob/aa0adddac414991b4f531a6c3b0634638ad8e337/aio/transforms/examples-package/services/region-parser.js#L71-L73

**Comment by petebacondarwin** [comment]
Rebased and fixed the commit message typo.

**Comment by petebacondarwin** [comment]
I can't approve my own PR - so I need one of @IgorMinar or @robwormald to do so.

**Review by IgorMinar — APPROVED** [review]
